app/views/users/show.html.erb

```diff
@@ -26,7 +26,7 @@
   <article>
     <% if current_user.id == @user.id %>
       <ul>
-        <h3>Recived Friend Requests:</h3>
+        <h3>Received Friend Requests:</h3>
         <% current_user.friend_requests.each do |friend| %>
           <li><%= friend.name %></li>
           <%= button_to "Accept Request", {:controller => "users", :action => "confirm", :friend_id => friend.id}, :method=>:post %>
```

Comment:
- In your friendship model, it's needed to implement this requirement:
 _when friendship is confirmed, an additional row with reversed user-friend pair is created_
 ``def confirm_friend``
``  self.update_attributes(confirmed: true)``
``  Friendship.create!(friend_id: self.user_id,``
   ``               user_id: self.friend_id,``
``                  confirmed: true)``
``end``

- thanks to this, we can find user friends only by user_id in friendships.
``has_many :confirmed_friendships, -> { where confirmed: true }, class_name: "Friendship"``
``has_many :friends, through: :confirmed_friendships``

-  User#friend_requests can be defined as association with SQL query instead of processing objects in application memory (as the database was created to store and query data it is always more efficient than ruby).
`` has_many :friend_requests, through: :inverted_friendships ``